sigstore-java/src/main/java/dev/sigstore/tuf/MetaStore.java

```diff
@@ -15,28 +15,25 @@
  */
 package dev.sigstore.tuf;
 
-import dev.sigstore.tuf.model.*;
+import dev.sigstore.tuf.model.Root;
+import dev.sigstore.tuf.model.SignedTufMeta;
+import dev.sigstore.tuf.model.TufMeta;
 import java.io.IOException;
 
-/** Defines the set of actions needed to support a local repository of TUF metadata. */
-public interface MutableTufStore extends TufStore {
-  /**
-   * Writes a TUF target to the local target store.
-   *
-   * @param targetName the name of the target file to write (e.g. ctfe.pub)
-   * @param targetContents the content of the target file as bytes
-   * @throws IOException if an error occurs
-   */
-  void storeTargetFile(String targetName, byte[] targetContents) throws IOException;
+/** Interface that defined a mutable meta store functionality. */
+public interface MetaStore extends MetaReader {
+
+  String getIdentifier();
 
   /**
-   * Generic method to store one of the {@link SignedTufMeta} resources in the local tuf store.
+   * Generic method to store one of the {@link SignedTufMeta} resources in the local tuf store. Do
+   * not use for Root role, use {@link #setRoot(Root)} instead.
```

Comment:
what happens if you do use it for Root?

Reply:
then you don't get N.root persisted (this is functionality that was already in there). I've just formalized it.